internal/application/ingresses.go

```diff
@@ -141,65 +154,41 @@ func SyncIngresses(ctx context.Context, cluster *kubernetes.Cluster, appRef mode
-	// name and has this Ingress as an owner.
-	name := names.IngressName(fmt.Sprintf("%s-%s", appRef.Name, route))
-	return &networkingv1.Ingress{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: name,
-			Annotations: map[string]string{
-				"traefik.ingress.kubernetes.io/router.entrypoints": "websecure",
-				"traefik.ingress.kubernetes.io/router.tls":         "true",
-				"kubernetes.io/ingress.class":                      "traefik",
-			},
-			Labels: map[string]string{
-				"app.kubernetes.io/component":  "application",
-				"app.kubernetes.io/managed-by": "epinio",
-				"app.kubernetes.io/name":       appRef.Name,
-				"app.kubernetes.io/created-by": username,
-				"app.kubernetes.io/part-of":    appRef.Org,
-			},
-		},
-		Spec: networkingv1.IngressSpec{
-			Rules: []networkingv1.IngressRule{
-				{
-					Host: route,
-					IngressRuleValue: networkingv1.IngressRuleValue{
-						HTTP: &networkingv1.HTTPIngressRuleValue{
-							Paths: []networkingv1.HTTPIngressPath{
-								{
-									Backend: networkingv1.IngressBackend{
-										Service: &networkingv1.IngressServiceBackend{
-											Name: names.ServiceName(appRef.Name),
-											Port: networkingv1.ServiceBackendPort{
-												Number: 8080,
-											},
-										},
-									},
-									Path:     "/",
-									PathType: &pathTypeImplementationSpecific,
-								},
-							},
-						},
-					},
-				},
-			},
-			TLS: []networkingv1.IngressTLS{
-				{
-					Hosts: []string{
-						route,
-					},
-					SecretName: name + "-tls",
-				},
-			},
-		},
+// completeIngress that takes an Ingress as created by the routes#ToIngress
+// method and fills in more data needed for Epinio.
+func completeIngress(ingress *networkingv1.Ingress, appRef models.AppRef, username string) *networkingv1.Ingress {
+	ingress.ObjectMeta.Annotations = map[string]string{
+		"traefik.ingress.kubernetes.io/router.entrypoints": "websecure",
+		"traefik.ingress.kubernetes.io/router.tls":         "true",
+		"kubernetes.io/ingress.class":                      "traefik",
 	}
+
+	ingress.ObjectMeta.Labels = map[string]string{
+		"app.kubernetes.io/component":  "application",
+		"app.kubernetes.io/managed-by": "epinio",
+		"app.kubernetes.io/name":       appRef.Name,
+		"app.kubernetes.io/created-by": username,
+		"app.kubernetes.io/part-of":    appRef.Org,
+	}
+
+	ingress.Spec.Rules[0].IngressRuleValue.HTTP.Paths[0].Backend =
+		networkingv1.IngressBackend{
+			Service: &networkingv1.IngressServiceBackend{
+				Name: names.ServiceName(appRef.Name),
+				Port: networkingv1.ServiceBackendPort{
+					Number: 8080,
+				}}}
+
+	ingress.Spec.TLS = []networkingv1.IngressTLS{{
+		Hosts: []string{ingress.Spec.Rules[0].Host},
+		// TODO:
+		// Should we re-use the same certificate if one exists for that domain?
+		// (e.g. other app on a different path). Who is the owner then?
+		// Is it ok to have 2 differnet certificates for the same domain?
```

Comment:
I think the only implication is that in the case of letsencrypt, we will issue more requests than we need to. But sharing certificates between ingresses when the host is similar requires more complicated handling. I'm not sure it's worth it.

Reply:
changed this comment to simply explain the situation